docs/contributing.md

```diff
@@ -1,65 +1,132 @@
-pipx uses [nox](https://pypi.org/project/nox/) for development, continuous integration testing, and automation.
+Thanks for your interest in contributing to pipx!
+
+
+## Dependencies
+pipx uses an automation tool called [nox](https://pypi.org/project/nox/) for development, continuous integration testing, and various tasks.
+
+`nox` defines tasks or "sessions" in `noxfile.py` which can be run with `nox -s SESSION_NAME`. Session names can be listed with `nox -l`.
+
+Install nox for pipx development:
+```
+pip install --user git+https://github.com/cs01/nox.git@5ea70723e9e6 nox
+```
+
+!!! note
+    A specific version of nox must be used for pipx development until [nox issue 233](https://github.com/theacodes/nox/issues/233) is resolved.
+
 
 ## Developing pipx
-pipx uses pipx to develop -- it's recursive! But don't worry, it's not that scary. You'll be up in running in no time. You'll also need `make` installed. If you don't have `make`, you can look at pipx's `makefile` to see the what the make targets correspond to).
+Clone pipx, then enter the root of the pipx source tree. Run this command to see which commands you can run with nox.
```

Comment:
Is this confusing seeing as we have a directory called `src/pipx`?  Maybe instead of `pipx source tree`, use `pipx repository` ?

Reply:
Ah yeah good call. I'll make that change.